pkg/apk/sign.go

```diff
@@ -32,8 +37,119 @@ import (
 	"github.com/chainguard-dev/go-apk/pkg/tarball"
 )
 
+var (
+	errNoPemBlock    = errors.New("no PEM block found")
+	errDigestNotSHA1 = errors.New("digest is not a SHA1 hash")
+	errNoPassphrase  = errors.New("key is encrypted but no passphrase was provided")
+	errNoRSAKey      = errors.New("key is not an RSA key")
+)
+
+type Signer interface {
+	Sign(data []byte) ([]byte, error)
+	KeyName() string
+}
+
+type Verifier interface {
+	Verify(data, signature []byte) error
+}
+
+type SignerVerifier interface {
+	Signer
+	Verifier
+}
+
+func NewKeySignerVerifier(privkeyFile, passphrase string) (SignerVerifier, error) {
+	pub, err := os.ReadFile(privkeyFile + ".pub")
+	if err != nil {
+		return nil, fmt.Errorf("read public key file: %w", err)
+	}
+	priv, err := os.ReadFile(privkeyFile)
+	if err != nil {
+		return nil, fmt.Errorf("read private key file: %w", err)
+	}
+	return &keySignerVerifier{
+		pubKey:      pub,
+		privKey:     priv,
+		privkeyFile: privkeyFile,
+		passphrase:  passphrase,
+	}, nil
+}
+
+func NewKeyVerifier(pubkey []byte) Verifier {
+	return &keySignerVerifier{
+		pubKey: pubkey,
+	}
+}
+
+type keySignerVerifier struct {
+	privkeyFile     string
+	privKey, pubKey []byte
+	passphrase      string
+}
+
+func (s *keySignerVerifier) KeyName() string {
+	return filepath.Base(s.privkeyFile)
+}
+
+func (s *keySignerVerifier) Sign(data []byte) ([]byte, error) {
+	if len(data) != sha1.Size {
+		return nil, errDigestNotSHA1
```

Comment:
Some KMS interfaces support taking the data and hashing itself, and some take the data pre-hashed.  Can we adapt the method name and argument to make it clearer that this is taking/signing the SHA1?